### ISSUE TITLE
qwen3.6-35B-A3B gets "dumbified" because of missing --jinja flag/internal jinja parsing.

### ISSUE THREAD

**Issue by ainoob2025** [lmstudio-ai/lmstudio-bug-tracker#2005]
**Which version of LM Studio?**
Example: LM Studio 0.4.15 (Build 2)

**Which operating system?**
Windows 11 Pro

**What is the bug?**
Always preferred LM Studio so far but when i thought about trying to finetune a model i tried Unsloth Studio.
I was wondering because the answers where much better in quality and also more precise  than in LM Studio. 
Exact same model, exact same settings. 
I tried it with hermes agent. 
With lm studio it startet exploring different approaches to the task, made a few tool calls, generated wrong toolcall syntax and after that immediately went in a tool loop. 
Same Task on Unsloth Studio, same inference settings (temp, top-p etc.). Result: 
ZERO tool call errors and it actually flew through the task in a structured, well organised way and delivered a really high quality output. 
After searching a while it seems to come down to the chat template or precisely using --jinja flag on llamacpp. 
I could reproduce the results of Unsloth with a direct llamacpp server instance with --jinja flag. Again Flawless execution.

it seems that the jinja parser in LM Studio actually BREAKS the Template and by that forces the model to go against its training data. Also LM studio Log says "enable thinging = 0" on Model load despite the model standard is "reasoning on". LlamaCPP says "Enable thinking = 1" when loading the model.

**Screenshots**

**Logs**
Hermes Agent chat:
Initializing agent...
────────────────────────────────────────

  ┊ 🌐 preparing browser_navigate…
  ┊ 📚 preparing skill_view…
  ┊ 🌐 navigate  github.com  0.4s [Unknown engine 'auto'. Supported engines: chr...]
  ┊ 📚 skill       0.0s [Skill '' not found.]
  ┊ 🌐 preparing browser_navigate…
  ┊ 🌐 navigate    0.3s [Unknown engine 'auto'. Supported engines: chr...]
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s [Invalid command: expected string, got NoneType]
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
  ┊ 💻 preparing terminal…
  ┊ 💻 $           0.0s
and so on, so I had to interrupt him here.

As statet Unsloth Studio had zero tool call errors (around 30-35). Here is the session output. sorry that it is baldy formated because i copied from terminal and i cant show including the perfect tool calls since they are not shown any more when resuming a session in Hermes.

Chat:
You: schau dir bitte https://github.com/HKUDS/ViMax an. Dort werden ausschliesslich teure cloud modelle verwendet. Wäre es theoretisch per config möglich das auf Lokale Modelle zu switchen?                                                                                                                             │
│   ◆ Hermes: Hier meine Analyse basierend auf dem echten Code (RTFM-Prinzip angewendet):                                                                         │
│                                                                                                                                                                 │
│             ## ViMax – Lokale Modelle: Theoretisch möglich?                                                                                                     │
│                                                                                                                                                                 │
│             **Kurze Antwort: Ja, aber nur beim Chat-Modell. Bild- und Video-Generatoren bleiben Cloud-abhängig.**                                               │
│                                                                                                                                                                 │
│             ### Was ich im Code gesehen habe:                                                                                                                   │
│                                                                                                                                                                 │
│             **1. Chat-Modul (wechselbar):**                                                                                                                     │
│                                                                                                                                                                 │
│             In `pipelines/idea2video_pipeline.py` wird der Chat-Modell so instantiiert:                                                                         │
│                                                                                                                                                                 │
│             ```python                                                                                                                                           │
│             chat_model_args = resolve_chat_model_config(config["chat_model"]["init_args"])                                                                      │
│             chat_model = init_chat_model(**chat_model_args)                                                                                                     │
│             ```                                                                                                                                                 │
│                                                                                                                                                                 │
│             `resolve_chat_model_config()` in `utils/provider_presets.py` baut auf **OpenAI-kompatiblen APIs** auf – `model_provider` wird nach Auflösung auf    │
│ `"openai"` normalisiert. Die drei Config-Parameter sind:                                                                                                        │
│                                                                                                                                                                 │
│             - `model` – Modellname                                                                                                                              │
│             - `base_url` – API-Endpunkt (Standard: OpenRouter)                                                                                                  │
│             - `api_key` – Schlüssel                                                                                                                             │
│                                                                                                                                                                 │
│             **=> Du kannst `base_url` auf jeden OpenAI-kompatiblen Endpunkt setzen:**                                                                           │
│                                                                                                                                                                 │
│             | Local-Option | Beispiel `base_url` |                                                                                                              │
│             |---|---|                                                                                                                                           │
│             | **Ollama** | `http://localhost:11434/v1` |                                                                                                        │
│             | **LM Studio** | `http://localhost:1234/v1` |                                                                                                      │
│             | **vLLM** | `http://localhost:8000/v1` |                                                                                                           │
│             | **KoboldCpp** | `http://localhost:5000/v1` |                                                                                                      │
│                                                                                                                                                                 │
│             Das `model`-Feld muss dann den Modellnamen enthalten, den der lokale Server referenziert (z.B. `qwen2.5:72b`, `llama3.1` etc.).                     │
│                                                                                                                                                                 │
│             **2. Image Generator (Cloud-Only):**                                                                                                                │
│                                                                                                                                                                 │
│             ```yaml                                                                                                                                             │
│             image_generator:                                                                                                                                    │
│               class_path: tools.ImageGeneratorNanobananaGoogleAPI                                                                                               │
│               # oder: tools.ImageGeneratorNanobananaYunwuAPI                                                                                                    │
│             ```                                                                                                                                                 │
│                                                                                                                                                                 │
│             Beide nutzen Google- oder Yunwu-Cloud-APIs – keine lokalen Alternativen eingebaut.                                                                  │
│                                                                                                                                                                 │
│             **3. Video Generator (Cloud-Only):**                                                                                                                │
│                                                                                                                                                                 │
│             ```yaml                                                                                                                                             │
│             video_generator:                                                                                                                                    │
│               class_path: tools.VideoGeneratorVeoGoogleAPI                                                                                                      │
│               # oder: tools.VideoGeneratorOmniYunwuAPI                                                                                                          │
│             ```                                                                                                                                                 │
│                                                                                                                                                                 │
│             Sowohl Google Veo als auch Yunwu Omni sind Cloud-Dienste.                                                                                           │
│                                                                                                                                                                 │
│             ### Praktisches Beispiel – Config für lokale Chat-Modelle:                                                                                          │
│                                                                                                                                                                 │
│             ```yaml                                                                                                                                             │
│             chat_model:                                                                                                                                         │
│               init_args:                                                                                                                                        │
│                 model: qwen2.5:72b                                                                                                                              │
│                 model_provider: openai                                                                                                                          │
│                 api_key: ollama                                                                                                                                 │
│                 base_url: http://localhost:11434/v1                                                                                                             │
│               max_requests_per_minute: 500                                                                                                                      │
│               max_requests_per_day: 2000                                                                                                                        │
│                                                                                                                                                                 │
│             image_generator:                                                                                                                                    │
│               class_path: tools.ImageGeneratorNanobananaGoogleAPI                                                                                               │
│               init_args:                                                                                                                                        │
│                 api_key: <DEIN_GOOGLE_API_KEY>                                                                                                                  │
│                                                                                                                                                                 │
│             video_generator:                                                                                                                                    │
│               class_path: tools.VideoGeneratorVeoGoogleAPI                                                                                                      │
│               init_args:                                                                                                                                        │
│                 api_key: <DEIN_GOOGLE_API_KEY>                                                                                                                  │
│                                                                                                                                                                 │
│             working_dir: .working_dir/idea2video                                                                                                                │
│             ```                                                                                                                                                 │
│                                                                                                                                                                 │
│             ### Fazit / Aufwand:                                                                                                                                │
│                                                                                                                                                                 │
│             | Teil | Local möglich? | Aufwand |                                                                                                                 │
│             |---|---|---|                                                                                                                                       │
│             | Chat-Modell (Screenwriter, Story-Planung, Character-Extraktion etc.) | **Ja** | Nur Config ändern |                                               │
│             | Image-Generator | **Nein** | Neue Klasse schreiben + Config |                                                                                     │
│             | Video-Generator | **Nein** | Neue Klasse schreiben + Config |

**Wichtig:** Der Chat-Modell braucht ein Modell mit guter multi-modalem Verständnis (da Bilder als Input kommen können) und starker Code-/JSON-Ausgabe (Scenes als JSON). Empfohlen: Qwen2.5-72B, Llama-3.1-70B, oder ähnlich starke Modelle. Kleinere Modelle (<8B) werden bei der Story- und  Scene-Planung wahrscheinlich schlechte Ergebnisse liefern. 

Die 3 Cloud-Module (Chat + Image + Video) sind im aktuellen Setup die Kosten-Treiber. Nur der Chat-Modul lässt sich ohne Code-Änderungen auf lokal umstellen.

**To Reproduce**
1. Use Hermes agent (nousresearch) configured for LM Studio usage with qwen3.6 35B-A3B model
2. give hermes a task where it has to do some webresearch and/or make a bunch of tool calls.
3. watch him trying different ways, making tool call errors and worst case end up looping.
4. give the same task again to a new session in Hermes agent configured to use the same model on a llamacpp server startet including flag --jinja
5. see how it works like a charm.

**note**
didn´t actually try but since i had similar tool calling issues with qwen3.5 9B i would assume this goes for the whole qwen3.5/3.6 family.

**Comment by KotPrezesOfficial** [comment]
Hi. Have you tried to use that  flag --jinja thing in the "my models" > Inference >Prompt template (template jinja) ?

**Comment by ainoob2025** [comment]
As far as i know there is no way to use llama-cpp flags directly with LM Studio on my Windows Desktop. The Template is set to Jinja by default and it shows the full template of the Model but it doesn´t work correctly. LM Studio seems to use some "MiniJinja" parser  and the original qwen3.5/3.6 templates got some arrays and things the miniJinja parser can´t handle (so i was told by gemini don´t know if its true). As described above: Loading the Model with LM Studio i have bad tool calling and "the way it works" seems "dumb". Using the Model in llama-cpp server directly with the --jinja flag it seems to be much smarter and when testing again with another task i gave Hermes Agent only one out of 34 Toolcalls failed and it didn´t repeat the mistake (on LM Studio it more often then not went into a tool loop and couldn´t recover himself). I had similar issues with qwen3.5 9B where tool calling also wasn´t really top notch. Until i saw how it worked and answered in Unsloth and later on using llama-ccp server i was just blaming it on "the small local and quantised model". In unsloth studio and llamacpp directly it actually felt like a "big one" for the first time, which impressed me and got me thinking since all parameters (load and inderence) were exactly the same. Since i am no dev i asked gemini and chatgpt and it came down to the template/how the template is actually used in LM Studio.
